src/Compilers/Core/Portable/Symbols/ITypeSymbol.cs

```diff
@@ -95,6 +96,21 @@ public interface ITypeSymbol : INamespaceOrTypeSymbol
         /// Must be a non-null interface property, method, or event.
         /// </param>
         ISymbol FindImplementationForInterfaceMember(ISymbol interfaceMember);
+
+        /// <summary>
+        /// True if the type is ref-like, meaning it follows rules similar to CLR by-ref variables. False if the type
+        /// is not ref-like or if the language has no concept of ref-like types.
+        /// </summary>
+        /// <remarks>
+        /// <see cref="Span{T}" /> is a commonly used ref-like type.
+        /// </remarks>
+        bool IsRefLikeType { get; }
```

Comment:
Consider having the comment mention ref structs.
I'm thinking maybe even the API should be called after ref structs (`IsRefType`).
On the other hand, the attribute which is emitted in PE is already called `IsByRefLikeTypeAttribute`, so that would be an argument to stick with that name for consistency.

Reply:
I was trying to follow the convention used by a lot of our public facing documents, like https://github.com/dotnet/csharplang/blob/master/proposals/csharp-7.2/span-safety.md